### DISCUSSION TITLE
Vercel Postgres Usage Patterns

### DISCUSSION THREAD

**Discussion by laruellef** [vercel/community#4492] [Feature Question]
### Question

Hi there,
I am not sure I understand how serverless Postgres really works, so I have a couple questions:
1. I have noticed that changes that I make manually to my db are not accessible to my app, unless I run a deploy, why is that?
2. Can this database be used to persist user data for example, does that survive between deployments?
Any insights appreciated

**Comment by amyegan** [comment]
Hi @laruellef! Can you walk me through your steps to manually update the db? I want to try them myself so I can see what's going wrong.

**Reply by laruellef** [reply]
Hi @amyegan ,
I use a postgres client named postico, but that shouldn't be important,
connect it with the connection params found in dashboard under the .env.local tab (specifically using the host, db, user and password)
I am able to interact with the db there, ie create, modify tables and add data.
however anything I commit does not make it to the app, even after browser refresh, unless I push the project.

**Reply by laruellef** [reply]
you should be able to replicate with just psql as well.

**Reply by laruellef** [reply]
so the steps would be:
1. create an app with a db, add some data to your db so you can display the data in the app.
2. deploy your app
3. now go and manually change the data in your db with a postgres client, using the connection params from dashboard .env.local tab. commit your changes.
4. refresh your app in your browser, but do not push or deploy it, and see if the data you added in step # 3 shows up (it doesn't for me)

**Reply by laruellef** [reply]
Hi there, any update on this?
